src/overlays/actors/ovl_En_Sekihi/z_en_sekihi.c

```diff
@@ -32,20 +45,150 @@ const ActorInit En_Sekihi_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+static s16 sObjectIds[] = { OBJECT_SEKIHIL, OBJECT_SEKIHIG, OBJECT_SEKIHIN, OBJECT_SEKIHIZ, OBJECT_ZOG };
+
+static Gfx* sOpaDLists[] = {
+    gGraveTriforceSymbolDL, gGraveGoronSymbolDL, gGraveKokiriSymbolDL, gTombSongOfSoaringDL, gMikauGraveDL,
+};
+
+static Gfx* sXluDLists[] = {
+    gSunSongTriforceSymbolDL, gSunSongGoronSymbolDL, gSunSongKokiriSymbolDL, gSekihizEmptyDL, gMikauGraveDirtDL,
+};
+
+static u16 sTextIds[] = { 0, 0, 0, 0, 0x1018 };
+
+void EnSekihi_Init(Actor* thisx, PlayState* play) {
+    EnSekihi* this = THIS;
+    s32 params = ENSIKIHI_GET_F(thisx);
+    s32 objectIndex;
+    s32 pad;
+
+    if ((params < SEKIHI_TYPE_0 || params >= SEKIHI_TYPE_MAX) || sOpaDLists[params] == SEKIHI_TYPE_0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        if ((params == SEKIHI_TYPE_4) && (((gSaveContext.save.skullTokenCount & 0xFFFF)) >= 30)) {
+            gSaveContext.save.weekEventReg[13] |= 0x20;
+        }
+        objectIndex = Object_GetIndex(&play->objectCtx, sObjectIds[params]);
+        if (objectIndex >= 0) {
+            this->objectIndex = objectIndex;
+        }
+        this->actionFunc = func_80A44DE8;
+        this->opaDList = sOpaDLists[params];
+        this->xluDList = sXluDLists[params];
+        this->dyna.actor.textId = sTextIds[params];
+        this->dyna.actor.focus.pos.y = this->dyna.actor.world.pos.y + 60.0f;
+        Actor_SetScale(&this->dyna.actor, 0.1f);
+    }
+}
+
+void EnSekihi_Destroy(Actor* thisx, PlayState* play) {
+    EnSekihi* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80A44DE8(EnSekihi* this, PlayState* play) {
+    CollisionHeader* colHeader = NULL;
+    s32 params;
+    CollisionHeader* sColHeader[] = {
```

Comment:
```suggestion
    CollisionHeader* colHeaders[] = {
```
Not static so no `s` as well as plural since it is an array.